packages/fx-core/src/core/middleware/projectMigrator.ts

```diff
@@ -524,6 +534,32 @@ async function getManifest(sourceManifestFile: string): Promise<TeamsAppManifest
   return await readJson(sourceManifestFile);
 }
 
+async function copyManifest(projectPath: string, fx: string, target: string) {
+  if (await fs.pathExists(path.join(projectPath, AppPackageFolderName))) {
+    await fs.copy(path.join(projectPath, AppPackageFolderName), target);
+  } else if (await fs.pathExists(path.join(fx, AppPackageFolderName))) {
+    // version <= 2.4.1
+    await fs.copy(path.join(fx, AppPackageFolderName), target);
+  } else {
+    // version <= 2.3.1
+    await fs.copy(path.join(fx, REMOTE_MANIFEST), path.join(target, REMOTE_MANIFEST));
+    const manifest: TeamsAppManifest = await getManifest(path.join(target, REMOTE_MANIFEST));
+    const color = (await fs.pathExists(path.join(fx, "color.png")))
+      ? "color.png"
+      : manifest.icons.color;
+    const outline = (await fs.pathExists(path.join(fx, "outline.png")))
```

Comment:
Should we check `manifest.icons.color` first?
```
    const color = (await fs.pathExists(manifest.icons.color))
      ? path.relative(fx, manifest.icons.color)
      : color.png)
```

Reply:
The reason for checking color.png is if user write manifest.icons.color in a wrong value and color.png exist, upgrade can still works.